ddtrace/monkey.py

```diff
@@ -56,127 +46,96 @@
+    :type raise_errors: bool
+    """
+    overrides = overrides or {}
+    integrations_to_install = DEFAULT_INTEGRATIONS.copy()
+    integrations_to_install.update(overrides)
+
+    for integration, enabled in integrations_to_install.items():
+        if not enabled:
+            log.info(
+                'install: skipping disabled integration "{}"'.format(integration)
+            )
+            continue
+        install(integration, raise_errors)
+
+
+def install(integration, raise_errors=False):
+    """Installs an integration using its ``patch()`` function.
+
+    :param integration: the integration to install
+    :type integration: str
+    :param raise_errors: whether or not to raise errors if they occur
+    :type raise_errors: bool
     """
-    modules = PATCH_MODULES.copy()
-    modules.update(patch_modules)
+    intmodname = integration_modname(integration)
 
-    patch(raise_errors=False, **modules)
+    try:
+        intmod = importlib.import_module(intmodname)
+    except ImportError:
+        log_msg = 'install: integration "{}" not found'.format(integration)
+        if raise_errors:
+            raise PatchException(log_msg)
+        else:
+            log.error(log_msg)
+            return
 
+    if not hasattr(intmod, 'patch'):
+        log.error('install: integration "{}" does not have patch attribute'.format(integration))
+        return
 
-def patch(raise_errors=True, **patch_modules):
-    """Patch only a set of given modules.
+    intmod.patch()
 
-    :param bool raise_errors: Raise error if one patch fail.
-    :param dict \**patch_modules: List of modules to patch.
 
-        >>> patch(psycopg=True, elasticsearch=True)
-    """
-    modules = [m for (m, should_patch) in patch_modules.items() if should_patch]
-    for module in modules:
-        if module in _PATCH_ON_IMPORT:
-            # If the module has already been imported then patch immediately
-            if module in sys.modules:
-                patch_module(module, raise_errors=raise_errors)
-
-            # Otherwise, add a hook to patch when it is imported for the first time
-            else:
-                # Use factory to create handler to close over `module` and `raise_errors` values from this loop
-                when_imported(module)(_on_import_factory(module, raise_errors))
-
-                # manually add module to patched modules
-                _PATCHED_MODULES.add(module)
-        else:
-            patch_module(module, raise_errors=raise_errors)
+def patch_all(**integrations):
+    """Patch all default enabled integrations.
 
-    patched_modules = get_patched_modules()
-    log.info(
-        "patched %s/%s modules (%s)",
-        len(patched_modules),
-        len(modules),
-        ",".join(patched_modules),
-    )
+    :param **integrations: override default enabled integrations.
+    :type integrations: dict
 
+    To patch all the default enabled integrations::
 
-def patch_module(module, raise_errors=True):
```

Comment:
We are removing this?

Reply:
Yeah it shouldn't be removed because I guess it is technically public (although our definition of public is not well defined 😝)